readthedocs/api/v2/views/integrations.py

```diff
@@ -290,19 +312,31 @@ def handle_webhook(self):
         if event in (GITHUB_CREATE, GITHUB_DELETE):
             return self.sync_versions(self.project)
 
-        if (
-            event == GITHUB_PULL_REQUEST and
-            self.data['action'] in [GITHUB_PULL_REQUEST_OPEN, GITHUB_PULL_REQUEST_SYNC]
-        ):
-            try:
-                identifier, verbose_name = self.get_external_version_data()
-                external_version = get_or_create_external_version(
-                    self.project, identifier, verbose_name
-                )
-                return self.get_response_push(self.project, [external_version.verbose_name])
-
-            except KeyError:
-                raise ParseError('Parameters "sha" and "number" are required')
+        if event == GITHUB_PULL_REQUEST and self.get_action():
```

Comment:
We shouldn't call `get_action` 3 times. It seems like we could remove this instance, and just get the value next then check it.

Reply:
we could do something like this `self.action = self.get_action()`